src/wireviz/DataClasses.py

```diff
@@ -119,23 +120,14 @@ def __post_init__(self) -> None:
                 raise Exception('Connectors with style set to simple may only have one pin')
             self.pincount = 1
 
-        if self.pincount is None:
-            if self.pinlabels:
-                self.pincount = len(self.pinlabels)
-            elif self.pins:
-                self.pincount = len(self.pins)
-            else:
+        if not self.pincount:
+            if not self.pins and not self.pinlabels:
                 raise Exception('You need to specify at least one, pincount, pins or pinlabels')
+            self.pincount = max(len(self.pins), len(self.pinlabels), len(self.pincolors))
```

Comment:
By moving the default `pincount` value computation 2 lines up, the `if` expression can be simplified:
```suggestion
            self.pincount = max(len(self.pins), len(self.pinlabels), len(self.pincolors))
            if not self.pincount:
                raise Exception('You need to specify at least one, pincount, pins, pinlabels, or pincolors')
```